packages/webapp/components/invite/Referral.tsx

```diff
@@ -0,0 +1,221 @@
+import { feature } from '@dailydotdev/shared/src/lib/featureManagement';
+import { GenericReferral } from '@dailydotdev/shared/src/lib/featureValues';
+import { ActionType } from '@dailydotdev/shared/src/graphql/actions';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import {
+  ApiErrorResult,
+  DEFAULT_ERROR,
+} from '@dailydotdev/shared/src/graphql/common';
+import { useAnalyticsContext } from '@dailydotdev/shared/src/contexts/AnalyticsContext';
+import { AnalyticsEvent } from '@dailydotdev/shared/src/lib/analytics';
+import {
+  ReferralCampaignKey,
+  useMedia,
+  useActions,
+  useToastNotification,
+} from '@dailydotdev/shared/src/hooks';
+import { downloadBrowserExtension } from '@dailydotdev/shared/src/lib/constants';
+import { FlexCentered } from '@dailydotdev/shared/src/components/utilities';
+import BrowsersIcon from '@dailydotdev/shared/icons/browsers.svg';
+import { mobileL, laptopL } from '@dailydotdev/shared/src/styles/media';
+import { Modal } from '@dailydotdev/shared/src/components/modals/common/Modal';
+import {
+  ModalKind,
+  ModalSize,
+} from '@dailydotdev/shared/src/components/modals/common/types';
+import CloseButton from '@dailydotdev/shared/src/components/CloseButton';
+import { anchorDefaultRel } from '@dailydotdev/shared/src/lib/strings';
+import { JoinPageProps } from './common';
+
+export function Referral({
+  referringUser,
+  redirectTo,
+  token,
+}: JoinPageProps): ReactElement {
+  const router = useRouter();
+  const genericReferral = useFeature(feature.genericReferral);
+  const { completeAction } = useActions();
+  const { trackEvent } = useAnalyticsContext();
+  const { displayToast } = useToastNotification();
+  const { user, refetchBoot, showLogin } = useAuthContext();
+  const isLaptopL = useMedia([laptopL.replace('@media ', '')], [true], false);
+  const {
+    mutateAsync: onAcceptMutation,
+    isLoading,
+    isSuccess,
+  } = useMutation(acceptFeatureInvitation, {
+    onSuccess: async () => {
+      await Promise.all([
+        completeAction(ActionType.AcceptedGenericReferral),
+        refetchBoot(),
+      ]);
+      router.push(redirectTo);
+    },
+    onError: (err: ApiErrorResult) => {
+      const message = err?.response?.errors?.[0]?.message;
+      displayToast(message ?? DEFAULT_ERROR);
+    },
+  });
+  const [isVideoOpen, setVideoOpen] = useState(false);
+
+  const onVideoModalClose = () => {
+    setVideoOpen(false);
+  };
+
+  const handleAcceptClick = () => {
+    const handleAccept = () =>
+      onAcceptMutation({
+        token,
+        referrerId: referringUser.id,
+        feature: ReferralCampaignKey.Generic,
+      });
+
+    if (!user) {
+      return showLogin({
+        trigger: AuthTriggers.GenericReferral,
+        options: {
+          onLoginSuccess: handleAccept,
+          onRegistrationSuccess: handleAccept,
+        },
+      });
+    }
+
+    // since in the page view, query params are tracked automatically,
```

Comment:
You double checked it on the actual event triggered? Assume so, but wanted to note it just to be safe.

Reply:
Actually, have been checking this a couple of times trying to understand when it should fire, and apparently it doesn't fire unless the user is logged in 🤔 
This was the same logic on the AISearchInvite, so it isn't firing there as well when it is supposed to.

`if (!user) {
      return showLogin({
        trigger: AuthTriggers.SearchReferral,
        options: {
          onLoginSuccess: handleAccept,
          onRegistrationSuccess: handleAccept,
        },
      });
    }`

Because of the above code it never goes to trackEvent unless there is a user, I believe that shouldn't be like this right?

Reply:
The login flow has its own tracking, part of the reason we don't want to send another event but only identify the trigger by passing the parameter `AuthTriggers.SearchReferral`.

Reply:
Cool, got it. Quick question though, if the user is logged in, so already has an account, do we know what should happen if user is invited to join and lands on the referral page?
In the code at the moment I believe that we still show the user the page, and if the user clicks the button in this case we would send the `accept invitation` event, but would this make sense if the user already has an account?
I might have missed something, but checking just in case

Reply:
Sorry, I just got back from the gym, answered in Slack now 🥳